controllers/module_reconciler.go

```diff
@@ -397,7 +407,7 @@ func (mrh *moduleReconcilerHelper) garbageCollect(ctx context.Context,
 	// Garbage collect old DaemonSets for which there are no nodes.
 	validKernels := sets.KeySet[string](mldMappings)
 
-	deleted, err := mrh.daemonAPI.GarbageCollect(ctx, existingDS, validKernels)
+	deleted, err := mrh.daemonAPI.GarbageCollect(ctx, mod, existingDS, validKernels)
```

Comment:
This is also deleting moduleLoader's DSs. I don't think we should do it inside `handelDevicePlugin` method.

Reply:
i think there is confusion here, we are doing it inside garbageCollection function

Reply:
I don't know why I though it was added to `handelDevicePlugin`... Please ignore this comment.